lib/database/mongodb.ts

```diff
@@ -29,15 +30,23 @@ export default (mPool: MongoClient): MongoDbService => {
     getWord: (wordId: ObjectId) =>
       getWordsCollection().findOne({ _id: new ObjectId(wordId) }),
 
-    getWords: async ({ page, itemsPerPage }: PaginatedWordsRequest) => {
-      const getPaginatedPromise = getWordsCollection()
-        .find()
+    getWords: async ({ page, itemsPerPage, tags }: PaginatedWordsRequest) => {
+      const getTagsFilter = () => {
+        if (tags === undefined) return {};
+        else if (tags.length === 0)
+          return { $or: [{ tags: undefined }, { tags: [] }] };
+        else return { tags: { $all: [...tags] } };
+      };
+
+      const filteredWordCollection = getWordsCollection().find(getTagsFilter());
```

Comment:
How about `filteredWordsCollection`? Or just `filteredWords`?

Reply:
Changed to `filteredWordsCollection`.